packages/discovery-react-components/src/utils/document/processDoc.ts

```diff
@@ -322,8 +315,16 @@ function setupSectionParser(
           parser.popState();
         }
       }
+      if (tagName === 'body') {
+        parser.popState();
+      }
     }
-  });
+  };
+
+  // Ensures the current section open tag is called first and then
+  // continues through tags inside of that section.
+  parser.pushState(actionState);
+  actionState.onopentag(sectionParser, sectionTagName, sectionTagAttrs);
```

Comment:
One thing I mentioned to you yesterday was that I didn't like having this bit of setup code separated from the rest by the large chunk of actionState` code. It would be very easy for someone reading this code to miss these 2 lines.

Instead, I suggest that we use function hoisting to our advantage, in order to keep the setup code together:

```javascript
  // track nested nodes of same tag name
  let stackCount = 1;
  let stackCount = 0;
  const openTagIndices = [0];
  const openTagIndices = [0];
  const sectionHtml = [openTagToString(sectionTagName, sectionTagAttrs, sectionChildBegin)];
  const sectionHtml: string[] = [];

  const actionState = getActionState();
  parser.pushSTate(actionState);

  // Ensures the current section open tag is called first and then
  // continues through tags inside of that section.
  actionState.onopentag(sectionParser, sectionTagName, sectionTagAttrs);

  function getActionState() {
    return {
      onopentag: ...
    }
  }
```